include/Tensor.hpp

```diff
@@ -493,6 +493,59 @@ namespace cytnx {
     // }
     //@}
 
+    // This mechanism is to remove the 'void' type from Type_list. Taking advantage of it
+    // appearing first ...
+    struct internal {
+      template <typename Variant>
+      struct exclude_first;
+
+      template <typename First, typename... Rest>
+      struct exclude_first<std::variant<First, Rest...>> {
+        using type = std::variant<Rest...>;
+      };
+    };  // internal
```

Comment:
Maybe this can be put in Type.hpp. And add `dtypes` which is a variant or a tuple that does not contain void, as you suggested before.

Reply:
Perhaps.  I would first look at how feasible it is to remove `void` from `Type_class`, rather than having two versions of the type list that differ only by one element in the public interface.